### PR TITLE
Updated to PMMP e388ac9

### PR DESCRIPTION
**Pull request by matcracker**
- Refactor Block::get() to BlockFactory::get()
- Refactor Block::{BLOCK_ID} to BlockLegacyIds::{BLOCK_ID}
- Refactor Item::fromString() to ItemFactory::fromString()
- Used GlobalLogger::get() instead of MainLogger::getLogger()

### CODE REVIEW

**Comment by Sandertv** [comment]
Looks good. Could you update the schematic submodule to the commit that merges your PR there?

Thanks!

File: src/BlockHorizons/BlockSniper/brush/BrushProperties.php [hunk 1]
```diff
@@ -156,7 +158,7 @@ public function getSoilBlocks() : array{
 	public function parseBlocks(string $data) : array{
 		$blocks = Parser::parse($data);
 		if(empty($blocks)){
-			return [Block::get(Block::AIR)];
+			return [BlockFactory::get(BlockLegacyIds::AIR)];
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/BrushProperties.php]: you can use `VanillaBlocks::AIR()` here now

File: src/BlockHorizons/BlockSniper/brush/TargetHighlight.php [hunk 2]
```diff
@@ -95,7 +96,7 @@ private function generateGeometry() : string{
 		}
 
 		$cubes = [];
-		foreach((new BlockEdgeIterator(Block::get(Block::AIR)))->getEdges() as $edge){
+		foreach((new BlockEdgeIterator(BlockFactory::get(BlockLegacyIds::AIR)))->getEdges() as $edge){
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/TargetHighlight.php]: same here

File: src/BlockHorizons/BlockSniper/brush/Type.php [hunk 3]
```diff
@@ -130,7 +132,7 @@ public function canBeExecutedAsynchronously() : bool{
 		try{
 			$this->brushBlocks = $this->properties->getBrushBlocks();
 		}catch(InvalidItemException $exception){
-			$this->brushBlocks = [Block::get(Block::AIR)];
+			$this->brushBlocks = [BlockFactory::get(BlockLegacyIds::AIR)];
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/Type.php]: and here

File: src/BlockHorizons/BlockSniper/brush/Type.php [hunk 4]
```diff
@@ -250,7 +252,7 @@ public function putBlock(Vector3 $pos, Block $block) : void{
 	 * @param Vector3 $pos
 	 */
 	public function delete(Vector3 $pos) : void{
-		$this->putBlock($pos, Block::get(Block::AIR));
+		$this->putBlock($pos, BlockFactory::get(BlockLegacyIds::AIR));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/Type.php]: and here

File: src/BlockHorizons/BlockSniper/brush/async/BlockSniperChunkManager.php [hunk 5]
```diff
@@ -62,6 +64,6 @@ public function getSide(int $x, int $y, int $z, int $side) : Block{
 			return $block;
 		}
 
-		return Block::get(Block::AIR);
+		return BlockFactory::get(BlockLegacyIds::AIR);
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/async/BlockSniperChunkManager.php]: and here

File: src/BlockHorizons/BlockSniper/brush/async/tasks/BrushTask.php [hunk 6]
```diff
@@ -116,7 +116,7 @@ private function blocks(Shape $shape, array $chunks) : Generator{
 
 			[$posX, $posY, $posZ] = [$vector3->x & 0x0f, $vector3->y, $vector3->z & 0x0f];
 			$combinedValue = $chunk->getFullBlock($posX, $posY, $posZ);
-			$block = Block::get($combinedValue >> 4, $combinedValue & 0xf);
+			$block = BlockFactory::get($combinedValue >> 4, $combinedValue & 0xf);
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/async/tasks/BrushTask.php]: `BlockFactory::fromFullBlock()` can be used here.

File: src/BlockHorizons/BlockSniper/brush/type/FreezeType.php [hunk 7]
```diff
@@ -20,23 +21,23 @@ class FreezeType extends Type{
 	public function fill() : Generator{
 		foreach($this->blocks as $block){
 			switch($block->getId()){
-				case Block::WATER:
-				case Block::FLOWING_WATER:
+				case BlockLegacyIds::WATER:
+				case BlockLegacyIds::FLOWING_WATER:
 					yield $block;
-					$this->putBlock($block, Block::get(Block::ICE));
+					$this->putBlock($block, BlockFactory::get(BlockLegacyIds::ICE));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/FreezeType.php]: `VanillaBlocks::ICE()` can be used here.

File: src/BlockHorizons/BlockSniper/brush/type/FreezeType.php [hunk 8]
```diff
@@ -20,23 +21,23 @@ class FreezeType extends Type{
...
 					break;
-				case Block::LAVA:
-				case Block::FLOWING_LAVA:
+				case BlockLegacyIds::LAVA:
+				case BlockLegacyIds::FLOWING_LAVA:
 					yield $block;
-					$this->putBlock($block, Block::get(Block::OBSIDIAN));
+					$this->putBlock($block, BlockFactory::get(BlockLegacyIds::OBSIDIAN));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/FreezeType.php]: `VanillaBlocks::OBSIDIAN()` can be used here.

File: src/BlockHorizons/BlockSniper/brush/type/FreezeType.php [hunk 9]
```diff
@@ -20,23 +21,23 @@ class FreezeType extends Type{
...
-				case Block::FIRE:
+				case BlockLegacyIds::FIRE:
 					yield $block;
 					$this->delete($block);
 					break;
-				case Block::ICE:
+				case BlockLegacyIds::ICE:
 					yield $block;
-					$this->putBlock($block, Block::get(Block::PACKED_ICE));
+					$this->putBlock($block, BlockFactory::get(BlockLegacyIds::PACKED_ICE));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/FreezeType.php]: `VanillaBlocks::PACKED_ICE()` can be used here.

File: src/BlockHorizons/BlockSniper/brush/type/HeatType.php [hunk 10]
```diff
@@ -22,37 +23,37 @@ class HeatType extends Type{
 	public function fill() : Generator{
 		foreach($this->blocks as $block){
 			switch($block->getId()){
-				case Block::PACKED_ICE:
-				case Block::ICE:
+				case BlockLegacyIds::PACKED_ICE:
+				case BlockLegacyIds::ICE:
 					yield $block;
-					$this->putBlock($block, Block::get(Block::WATER));
+					$this->putBlock($block, BlockFactory::get(BlockLegacyIds::WATER));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/HeatType.php]: `VanillaBlocks::WATER()`

File: src/BlockHorizons/BlockSniper/brush/type/HeatType.php [hunk 11]
```diff
@@ -22,37 +23,37 @@ class HeatType extends Type{
...
 						$this->delete($block);
 					}
 					break;
-				case Block::GRASS:
+				case BlockLegacyIds::GRASS:
 					$random = random_int(0, 8);
 					if($random === 0){
 						yield $block;
-						$this->putBlock($block, Block::get(Block::DIRT));
+						$this->putBlock($block, BlockFactory::get(BlockLegacyIds::DIRT));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/HeatType.php]: `VanillaBlocks::DIRT()`

File: src/BlockHorizons/BlockSniper/brush/type/HeatType.php [hunk 12]
```diff
@@ -22,37 +23,37 @@ class HeatType extends Type{
...
 					}elseif($random === 1){
 						yield $block;
-						$this->putBlock($block, Block::get(Block::DIRT, 1));
+						$this->putBlock($block, BlockFactory::get(BlockLegacyIds::DIRT, 1));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/HeatType.php]: `VanillaBlocks::COARSE_DIRT()`

File: src/BlockHorizons/BlockSniper/brush/type/HeatType.php [hunk 13]
```diff
@@ -22,37 +23,37 @@ class HeatType extends Type{
...
 					}
 					break;
 				case $block instanceof Flower || $block instanceof DoublePlant || $block instanceof TallGrass:
 					yield $block;
-					$this->putBlock($block, Block::get(Block::DEAD_BUSH));
+					$this->putBlock($block, BlockFactory::get(BlockLegacyIds::DEAD_BUSH));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/HeatType.php]: `VanillaBlocks::DEAD_BUSH()`

File: src/BlockHorizons/BlockSniper/brush/type/SnowConeType.php [hunk 14]
```diff
@@ -26,12 +27,12 @@ public function fill() : Generator{
 			if(!($block instanceof Flowable) && !($block instanceof Air) && !($block instanceof SnowLayer)){
 				$topBlock = $this->side($block, Facing::UP);
 				if($topBlock instanceof Air || $topBlock instanceof SnowLayer){
-					if($topBlock->getMeta() < 7 && $topBlock->getId() === Block::SNOW_LAYER){
+					if($topBlock->getMeta() < 7 && $topBlock->getId() === BlockLegacyIds::SNOW_LAYER){
 						yield $topBlock;
-						$this->putBlock($topBlock, Block::get(Block::SNOW_LAYER, $topBlock->getMeta() + 1));
+						$this->putBlock($topBlock, BlockFactory::get(BlockLegacyIds::SNOW_LAYER, $topBlock->getMeta() + 1));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/SnowConeType.php]: `VanillaBlocks::SNOW_LAYER()`

File: src/BlockHorizons/BlockSniper/brush/type/SnowConeType.php [hunk 15]
```diff
@@ -26,12 +27,12 @@ public function fill() : Generator{
...
 					}elseif(!($topBlock instanceof SnowLayer)){
 						yield $topBlock;
-						$this->putBlock($topBlock, Block::get(Block::SNOW_LAYER));
+						$this->putBlock($topBlock, BlockFactory::get(BlockLegacyIds::SNOW_LAYER));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/SnowConeType.php]: `VanillaBlocks::SNOW_LAYER()`

File: src/BlockHorizons/BlockSniper/brush/type/WarmType.php [hunk 16]
```diff
@@ -16,17 +17,17 @@ class WarmType extends Type{
...
-				case Block::SNOW_LAYER:
+				case BlockLegacyIds::SNOW_LAYER:
 					yield $block;
 					$this->delete($block);
 					break;
-				case Block::PACKED_ICE:
+				case BlockLegacyIds::PACKED_ICE:
 					yield $block;
-					$this->putBlock($block, Block::get(Block::ICE));
+					$this->putBlock($block, BlockFactory::get(BlockLegacyIds::ICE));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/WarmType.php]: `VanillaBlocks::ICE()`

File: src/BlockHorizons/BlockSniper/brush/type/WarmType.php [hunk 17]
```diff
@@ -16,17 +17,17 @@ class WarmType extends Type{
 	public function fill() : Generator{
 		foreach($this->blocks as $block){
 			switch($block->getId()){
-				case Block::ICE:
+				case BlockLegacyIds::ICE:
 					yield $block;
-					$this->putBlock($block, Block::get(Block::WATER));
+					$this->putBlock($block, BlockFactory::get(BlockLegacyIds::WATER));
```
Comment (dktapps) [inline src/BlockHorizons/BlockSniper/brush/type/WarmType.php]: `VanillaBlocks::WATER()`

**Comment by matcracker** [comment]
@dktapps Yes, I was so focused to fix the errors without thinking the new VanillaBlocks feature. I'll fix everything later.

**Pull request merged**